Rakefile

```diff
@@ -151,6 +151,24 @@ namespace :build do
         HipChat.log 'Migrating <b>dashboard</b> database...'
         RakeUtils.rake 'db:migrate'
 
+        # Update the schema cache file, except for production which always uses the cache.
+        schema_cache_file = dashboard_dir('db/schema_cache.dump')
+        unless rack_env?(:production)
+          RakeUtils.rake 'db:schema:cache:dump'
+          if RakeUtils.file_changed_from_git?(schema_cache_file)
+            # Staging is responsible for committing the authoritative schema cache dump.
+            if rack_env?(:staging)
+              RakeUtils.system 'git', 'add', schema_cache_file
```

Comment:
ah, I think add is necessary for new files but not for existing. So you do
need it if you want this script to be able to handle the file not existing.

On Wed, Feb 3, 2016 at 2:58 PM, philbogle <notifications@github.com> wrote:

> In Rakefile
> <https://github.com/code-dot-org/code-dot-org/pull/6586#discussion_r51802600>
> :
>
> > @@ -151,6 +151,24 @@ namespace :build do
> >          HipChat.log 'Migrating <b>dashboard</b> database...'
> >          RakeUtils.rake 'db:migrate'
> >
> > +        # Update the schema cache file, except for production which always uses the cache.
> > +        schema_cache_file = dashboard_dir('db/schema_cache.dump')
> > +        unless rack_env?(:production)
> > +          RakeUtils.rake 'db:schema:cache:dump'
> > +          if RakeUtils.file_changed_from_git?(schema_cache_file)
> > +            # Staging is responsible for committing the authoritative schema cache dump.
> > +            if rack_env?(:staging)
> > +              RakeUtils.system 'git', 'add', schema_cache_file
>
> I think the add is still needed. An experiment suggests that you still
> need to add the file before you can commit it, even when you specify it on
> the command line for the commit:
>
> $ echo "hello" > foo.txt
> $ git commit -m "add foo" foo.txt
> error: pathspec 'foo.txt' did not match any file(s) known to git.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/code-dot-org/code-dot-org/pull/6586/files#r51802600>.
>